base/permuteddimsarray.jl

```diff
@@ -12,7 +12,7 @@ struct PermutedDimsArray{T,N,perm,iperm,AA<:AbstractArray} <: AbstractArray{T,N}
     function PermutedDimsArray{T,N,perm,iperm,AA}(data::AA) where {T,N,perm,iperm,AA<:AbstractArray}
         (isa(perm, NTuple{N,Int}) && isa(iperm, NTuple{N,Int})) || error("perm and iperm must both be NTuple{$N,Int}")
         isperm(perm) || throw(ArgumentError(string(perm, " is not a valid permutation of dimensions 1:", N)))
-        all(map(d->iperm[perm[d]]==d, 1:N)) || throw(ArgumentError(string(perm, " and ", iperm, " must be inverses")))
+        all(ntuple(d->iperm[perm[d]]==d, Val{N}())) || throw(ArgumentError(string(perm, " and ", iperm, " must be inverses")))
```

Comment:
Do we even need any collection at all? We could simply pass a predicate to `all`:
```julia
all(d -> iperm[perm[d]] == d, 1:N)
```
(I might be overlooking something)

Reply:
It looks better, but `all(...` is usually slower than `all(ntuple...` because of the short circuit.

Reply:
Hmm its possible if people are permuting 50 dimensions your version is better? I rarely do more than five or six so I didn't consider that. At some length the compiler might give up on `ntuple`.

Reply:
I doubt that this line will impact runtime performance in a real-world scenario. Perhaps it's best to use the predicate form, where a method of `all` wouldn't need to be compiled for every tuple size.

Reply:
Ok I switched it to just use `all`